src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20211111103109_EES2878UpdateReleaseTypeTitles.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Migrations.ContentMigrations
+{
+    public partial class EES2878UpdateReleaseTypeTitles : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql("UPDATE ReleaseTypes SET Title = 'Ad Hoc Statistics' WHERE Id = '1821abb8-68b0-431b-9770-0bea65d02ff0';");
```

Comment:
We've now got a mixture of the ReleaseTypes reference data being managed within ContentDbContext and outside of its control as I think two new release types have recently been introduced without updating the definition that already exists there. See `modelBuilder.Entity<ReleaseType>().HasData` where they are defined.

We should have either introduced the new types to that, and be changing these titles by updating that definition, or we should have got rid of it and managed the unwanted migration that it would have generated removing them, and then this PR would have been acceptable.

I think there's an issue with removing the definition altogether that if the migrations are baselined, the table will be empty if starting up with a fresh db. As the application code expects these types to exist I would be tempted to keep the definition in code.

Reply:
I added the two new release types to ContentDbContext, and when I created a migration it tries to insert the new data. But this wouldn't work because those rows were already added in the previous migration. So I've manually added lines to delete the new release types first with a comment referencing the migration where they were created.